### PR TITLE
Support Hidden Application Attributes

### PR DESCRIPTION
**Pull request by Arshardh**
With this new feature, in addition to defining whether an attribute is required, the user can also define whether that attribute should be hidden. When an attribute is hidden, it will not be shown in the store UI, and this attribute should not be provided when creating an application through the rest API.
If the hidden attribute is defined as required, we add the default value defined will be added as the value for the hidden attribute.
If a required value is not provided when creating an application (in case this is done via the store REST APIs), and a default value is given, we add the default value, and if a default value is not given, we respond with a Bad Request error.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java [hunk 1]
```diff
@@ -323,28 +323,37 @@ private void readChildElements(OMElement serverConfig,
...
+            } else if (APIConstants.WorkflowConfigConstants.WORKFLOW.equals(localName)) {
                 setWorkflowProperties(serverConfig);
             } else if (APIConstants.ApplicationAttributes.APPLICATION_ATTRIBUTES.equals(localName)) {
                 Iterator iterator = element.getChildrenWithLocalName(APIConstants.ApplicationAttributes.ATTRIBUTE);
                 while (iterator.hasNext()) {
                     OMElement omElement = (OMElement) iterator.next();
                     Iterator attributes = omElement.getChildElements();
                     JSONObject jsonObject = new JSONObject();
-                    while (attributes.hasNext()) {
+                    boolean isHidden = Boolean.parseBoolean(omElement.getAttributeValue(new QName("hidden")));
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java]: hidden and required values are also can be moved to constant

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java [hunk 2]
```diff
@@ -323,28 +323,37 @@ private void readChildElements(OMElement serverConfig,
...
                     JSONObject jsonObject = new JSONObject();
-                    while (attributes.hasNext()) {
+                    boolean isHidden = Boolean.parseBoolean(omElement.getAttributeValue(new QName(APIConstants.ApplicationAttributes.HIDDEN)));
+                    boolean isRequired =
+                            Boolean.parseBoolean(omElement.getAttributeValue(new QName(APIConstants.ApplicationAttributes.REQUIRED)));
+                    jsonObject.put(APIConstants.ApplicationAttributes.HIDDEN, isHidden);
+                    while(attributes.hasNext()) {
                         OMElement attribute = (OMElement) attributes.next();
-                        if (attribute.getLocalName().equals("Name")) {
+                        if(attribute.getLocalName().equals(APIConstants.ApplicationAttributes.NAME)){
```
Comment (dinusha92) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java]: formatting issue.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java [hunk 3]
```diff
@@ -323,28 +323,37 @@ private void readChildElements(OMElement serverConfig,
...
                             jsonObject.put(APIConstants.ApplicationAttributes.ATTRIBUTE, attribute.getText());
-                        } else if (attribute.getLocalName().equals("Description")) {
+                        }
+                        else if(attribute.getLocalName().equals(APIConstants.ApplicationAttributes.DESCRIPTION)) {
```
Comment (dinusha92) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java]: Formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java [hunk 4]
```diff
@@ -323,28 +323,37 @@ private void readChildElements(OMElement serverConfig,
...
                             jsonObject.put(APIConstants.ApplicationAttributes.DESCRIPTION, attribute.getText());
                         }
+                        else if(attribute.getLocalName().equals(APIConstants.ApplicationAttributes.DEFAULT) && isRequired) {
```
Comment (dinusha92) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java]: Formatting issue

**Pull request merged**